api/src/main/java/jakarta/data/metamodel/restrict/BasicRestrictionRecord.java

```diff
@@ -50,20 +56,12 @@ public BasicRestriction<T> negate() {
      */
     @Override
     public String toString() {
-        String comparisonString = comparison.asQueryLanguage();
-        String valueString = value.toString();
-        StringBuilder builder = new StringBuilder(
-                attribute.length() +
-                comparisonString.length() +
-                valueString.length() +
-                4); // number of additional characters that might be appended
-        builder.append(attribute).append(' ')
-               .append(comparisonString).append(' ');
-        if (value instanceof CharSequence) {
-            builder.append('"').append(valueString).append('"');
-        } else {
-            builder.append(valueString);
-        }
-        return builder.toString();
+        final Operator comparison = comparison();
+        final String op = comparison.asQueryLanguage();
+        return switch (comparison.arity()) {
+            case 1 -> attribute + ' ' + op;
+            case 2,3 -> attribute + ' ' + op + ' ' + constraint;
```

Comment:
Goes along with other comments
```suggestion
            case 2,3 -> attribute + ' ' + constraint;
```